File: cobots_core/src/cobots_datatypes/location.py
from waypoint import Waypoint


class Location(Waypoint):

    def __init__(self, position=None, orientation=None, joints=None, type='',
                 name='', uuid=None, parent=None, append_type=True):
        super(Waypoint,self).__init__(
            position=position,
            orientation=orientation,
            joints=joints,
            type='location.'+type if append_type else type,
            name=name,
            uuid=uuid,
            parent=parent,
            append_type=append_type)

File: evd_unity_sim_app/Assets/Application/_Scripts/evd_script (environment-simplified)/data/Location.cs
using System.Collections;
using System.Collections.Generic;


namespace EvD
{
    namespace Data
    {

        [System.Serializable]
        public class Location : Waypoint
        {

            /*
            * Constructors
            */

            public Location(Position position = null, Orientation orientation = null, double[] joints = null, 
                            string type="", string name = "", string uuid = null, Node parent = null, 
                            bool appendType=true) 
            : base(position,orientation,joints, appendType ? "location." + type : type, name, uuid,parent,appendType) { }

            public new static Location FromDict(Dictionary<string, object> dct)
            {
                return new Location(
                    position: Position.FromDict((Dictionary<string, object>)dct["position"]),
                    orientation: Orientation.FromDict((Dictionary<string, object>)dct["orientation"]),
                    joints: (double[])dct["joints"],
                    type: (string)dct["type"],
                    name: (string)dct["name"],
                    uuid: (string)dct["uuid"],
                    appendType: false
                );
            }
        }

    }
}